india_compliance/gst_india/utils/gstin_info.py

```diff
@@ -23,20 +23,28 @@
 
 
 @frappe.whitelist()
-def get_gstin_info(gstin):
+def get_gstin_info(gstin, *, throw_error=True):
     if not frappe.get_cached_doc("User", frappe.session.user).has_desk_access():
         frappe.throw(_("Not allowed"), frappe.PermissionError)
 
     validate_gstin(gstin)
     response = get_archived_gstin_info(gstin)
 
     if not response:
-        response = PublicAPI().get_gstin_info(gstin)
-        frappe.enqueue(
-            "india_compliance.gst_india.doctype.gstin.gstin.create_or_update_gstin_status",
-            queue="long",
-            response=response,
-        )
+        try:
+            response = PublicAPI().get_gstin_info(gstin)
+            frappe.enqueue(
+                "india_compliance.gst_india.doctype.gstin.gstin.create_or_update_gstin_status",
+                queue="long",
+                response=response,
+            )
+        except Exception as exc:
+            if throw_error:
+                raise exc
```

Comment:
```diff
- raise exc
+ raise
```


@vorasmit we can ignore all errors here, the setup even if it fails can be re-done, however if it causes seutp-wizard to fail then site kinda bricks :sweat_smile:

Reply:
This API is called at all places. Even in the Party Quick Entry Form.

Refer: india_compliance/setup_wizard.py
It's with `throw_error = False`. Hence no errors would be raised for setup wizard.